src/utils/types.ts

```diff
@@ -300,7 +310,12 @@ export interface UploadedFile {
   name: string; // name is the name of the file in blob storage
 }
 
+// BaseEmailTemplateData holds the required values for the base email
+export interface BaseEmailTemplateData {
+  name: string;
+}
+
 export interface Email {
-  data: Record<string, unknown>; // data contains the the variables for the template
+  data: Record<string, unknown> & BaseEmailTemplateData; // data contains the the variables for the template
```

Comment:
Assuming every Email extends the BaseTemplate, they all need a "name" value supplied.